nmostesting/suites/BCP00301Test.py

```diff
@@ -265,7 +265,7 @@ def test_07(self, test):
 
         try:
             context = ssl.create_default_context(cafile=CONFIG.CERT_TRUST_ROOT_CA)
-            hostname = self.apis[SECURE_API_KEY]["hostname"]
+            hostname = self.apis[SECURE_API_KEY]["hostname"].rstrip('.')

```

Comment:
you need to do the same trick here and pass the stripped hostname as the `server_hostname` but the unstripped one as hostname in `connect`, I believe.